pkg/svcat/service-catalog/plan.go

```diff
@@ -60,39 +59,52 @@ type Plan interface {
 }
 
 // RetrievePlans lists all plans defined in the cluster.
-func (sdk *SDK) RetrievePlans(opts RetrievePlanOptions) ([]Plan, error) {
+func (sdk *SDK) RetrievePlans(classID string, opts RetrievePlanScopedOptions) ([]Plan, error) {
+	plans, err := sdk.retrievePlansByListOptions(opts, v1.ListOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	if classID == "" {
+		return plans, nil
+	}
+
+	var filtered []Plan
+	for _, p := range plans {
+		if p.GetClassID() == classID {
+			filtered = append(filtered, p)
+		}
+	}
+
+	return filtered, nil
+}
+
+func (sdk *SDK) retrievePlansByListOptions(scopedOpts RetrievePlanScopedOptions, listOpts v1.ListOptions) ([]Plan, error) {
 	var plans []Plan
 
-	if opts.Scope.Matches(ClusterScope) {
-		csp, err := sdk.ServiceCatalog().ClusterServicePlans().List(v1.ListOptions{})
+	if scopedOpts.Scope.Matches(ClusterScope) {
+		csp, err := sdk.ServiceCatalog().ClusterServicePlans().List(listOpts)
```

Comment:
This is my current concern with this approach. Imagine the following scenario, system has a Namespace Scoped plan named "myplan" and user wants to describe it but it does use AllScope instead of NamespaceScope on command flags. This function will first call List() from ClusterServicePlans(), since AllScope does match ClusterScope, which will probably fail with "Plan not found". In that case the function will return, what is wrong since the plan does exist. But, at the same time, I am not sure we can just ignore the error here, since it can be something relevant for user. @carolynvs, any suggestion on how to handle this?

Reply:
Using the all scope with a command that operates on a single resource (i.e. get FOO or describe FOO) is invalid. The cli will validate for that.

We can also put a quick check at the top of this function and return an error if all is specified. What do you think?

Reply:
If you believe it would be good I can add, but if such a scenario is validated on the CLI I think should be fine. Thanks a lot for clarifying that :)

Reply:
Eventually it would be good to not assume that people are using the cli. Maybe someone would find it helpful to use our pkg/svcat lib directly. So seems like it's worth adding that as a quick check to avoid unexpected behavior.

Reply:
I see that you added validations for the AllScope, but missed this function. Would you please add that check to this function as well?